src/main/java/seedu/address/logic/parser/patient/DeletePatientCommandParser.java

```diff
@@ -13,18 +16,41 @@
  */
 public class DeletePatientCommandParser implements Parser<DeletePatientCommand> {
 
+    /**
+     * Used for the separation of force delete format and patient index.
+     */
+    private static final Pattern FORCE_DELETE_FORMAT = Pattern.compile("(?<forceDelete>\\D+)(?<patientIndex>.*)");
+
     /**
      * Parses the given {@code String} of arguments in the context of the DeletePatientCommand
      * and returns a DeletePatientCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
     public DeletePatientCommand parse(String args) throws ParseException {
+        final Matcher forceDeleteMatcher = FORCE_DELETE_FORMAT.matcher(args.trim());
+        boolean isForceDelete = false;
+        String indexToParse = args;
+
         try {
-            Index index = ParserUtil.parseIndex(args);
-            return new DeletePatientCommand(index);
+            if (forceDeleteMatcher.matches()) {
+                final String forceDelete = forceDeleteMatcher.group("forceDelete").trim();
+                final String patientIndex = forceDeleteMatcher.group("patientIndex");
+
+                if (forceDelete.equals("--force")) {
+                    isForceDelete = true;
+                    indexToParse = patientIndex;
+                }
+            }
+
```

Comment:
i believe this portion of the code doesn't have to be in a `try`?
i think its cleaner to move it out of the try if it doesn't have to be inside.
u can then handle the `isForceDelete` cleaner as well (ie maybe in the else)

Reply:
Yes I think it might not need to be in the `try` block, will edit later. The reason why I wrote `isForceDelete` is to avoid the possibility of `isForceDelete` being not instantialized. Another way is shifting the try block inside the first if statement, which may be a little ugly, almost arrowhead style. What are your thoughts?

Reply:
yeah that does sound abit ugly.
i'd prefer just the 1 line (or multiple lines) which causes the exception to be within the try, so it makes it clearer for readers too